backend/src/load.rs

```diff
@@ -0,0 +1,404 @@
+use std::collections::{HashMap, HashSet};
+use std::cmp::{Ordering, max};
+use std::io::Read;
+
+use chrono::{Duration, NaiveDateTime};
+use json::{self, JsonValue};
+
+use util::start_idx;
+use errors::*;
+
+const WEEKS_IN_SUMMARY: usize = 12;
+
+/// Contains the data loaded from the file system.
+/// Useful to avoid passing around each individual field, instead passing the
+/// entire struct.
+pub struct ServerData {
```

Comment:
SO is this the data we read in from the GH repo? If so, then maybe `InputData` or `RawData` is a better name